### PR TITLE
ci: improve coverage of package config

### PR DESCRIPTION
**Pull request by GregoryAlbouy**
## Changes

- Rename some existing tests
- Refactor some existing tests
- Add some doc comments
- Change Global.Validate error messages
- ...

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -150,54 +147,54 @@ func (cfg *Global) overrideHeader(newHeader http.Header) {
 
 // Validate returns the config and a not nil ErrInvalid if any of the fields provided by the user is not valid
```
Comment (c1-ra) [inline config/config.go:148]: `ErrInvalid` became an `InvalidConfigError`!

File: config/config_test.go [hunk 2]
```diff
@@ -13,85 +15,69 @@ import (
...
-		err := cfg.Validate()
-		if err != nil {
-			t.Errorf("valid configuration not considered as such")
+		if err := cfg.Validate(); err != nil {
+			t.Errorf("unexpected error: %v", err)
 		}
 	})
 
-	t.Run("test invalid configuration returns ErrInvalid error with correct messages", func(t *testing.T) {
+	t.Run("return cumulated errors if config in invalid", func(t *testing.T) {
```
Comment (c1-ra) [inline config/config_test.go:40]: A small typo here : ~~in~~ -> is

File: config/config.go (hunk 1 above)
Comment (c1-ra) [inline config/config.go:148]: `ErrInvalid` became `InvalidConfigError`!

**Comment by c1-ra** [comment]
[question]
Looks good, I ust have one question (concerning what had been done before, but also errors):
Since providing an invalid Url to `withURL()` results in setting an empty url, when we use `Validate()` on it, won't the message be "missing url" instead of "invalid url"?

**Review by moreirathomas — APPROVED** [review]

**Comment by GregoryAlbouy** [comment]
> [question] Looks good, I just have one question (concerning what had been done before, but also errors, so it could be changer here): Since providing an invalid Url to `withURL()` results in setting an empty url, when we use `Validate()` on it, won't the message be "missing url" instead of "invalid url"?

It results in a weird mix: the result is "invalid url" because we check for `nil`, not _empty_.
But because the resulting url is empty, the error message is not ideal, like:
```txt
url (""): invalid url
     ^^ not the input url
```

I suggest we leave this aside for now 😛

**Review by c1-ra — APPROVED** [review]

**Comment by c1-ra** [comment]
I agree, it is far from critical, let's move forward!

**Pull request merged**